Bottom: 4c4025fc0826219a4d90f9c1b214bfcfef9146a5
Top:    b0180a3442c84191050bdb9eea803d6aed85cc93
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 15:54:12 +0100

Add magic flag to stop pour clipping for off-screen buffers

















---

diff --git a/src/create.c b/src/create.c
index c228c87..a2b1ed6 100644
--- a/src/create.c
+++ b/src/create.c
@@ -151,6 +151,7 @@ CreateNewPCB (bool SetDefaultNames)
   ptr = (PCBTypePtr)calloc (1, sizeof (PCBType));
   ptr->Data = CreateNewBuffer ();
   ptr->Data->pcb = (PCBTypePtr) ptr;
+  ptr->Data->ClipPours = 1;
 
   ptr->ThermStyle = 4;
   ptr->IsleArea = 2.e8;
diff --git a/src/global.h b/src/global.h
index 7c0d2ed..43ab6f6 100644
--- a/src/global.h
+++ b/src/global.h
@@ -405,6 +405,7 @@ typedef struct			/* holds all objects */
    *rat_tree;
   struct PCBType *pcb;
   LayerType Layer[MAX_LAYER + 2];	/* add 2 silkscreen layers */
+  int ClipPours;
 } DataType, *DataTypePtr;
 
 typedef struct			/* holds drill information */
diff --git a/src/pour.c b/src/pour.c
index 593e5b0..2bb9918 100644
--- a/src/pour.c
+++ b/src/pour.c
@@ -1389,6 +1389,8 @@ PlowPours (DataType * Data, int type, void *ptr1, void *ptr2,
 void
 RestoreToPours (DataType * Data, int type, void *ptr1, void *ptr2)
 {
+  if (!Data->ClipPours)
+    return;
   if (type == POUR_TYPE)
     {
 #warning FIXME Later: Why do we need to do this?
@@ -1401,6 +1403,8 @@ RestoreToPours (DataType * Data, int type, void *ptr1, void *ptr2)
 void
 ClearFromPours (DataType * Data, int type, void *ptr1, void *ptr2)
 {
+  if (!Data->ClipPours)
+    return;
   if (type == POUR_TYPE)
     {
 #warning FIXME Later: Why do we need to do this?
